AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py

```diff
@@ -101,30 +101,38 @@ def enter_game_mode(msgtuple_success_fail: Tuple[str, str]) -> None:
-        :param sv_default_player_spawn_asset: The path to the network player prefab that will be automatically spawned upon entering gamemode.  The engine default is "prefabs/player.network.spawnable" 
+        Looks for an expected line in a list of tracer log lines
+        :param window: The log's window name. For example, logs printed via script-canvas use the "Script" window. 
+        :param expected_message: The log message to search. 
+        :param print_infos: A list of PrintInfos collected by Tracer to search. Example options: your_tracer.warnings, your_tracer.errors, your_tracer.asserts, or your_tracer.prints 
 
-        :return: None
+        :return: True if the message is found, otherwise false.
         """
+        for printInfo in print_infos:
+            if printInfo.window == window.strip() and printInfo.message.strip() == expected_message:
+                return True
+        return False
 
-        # looks for an expected line in a list of tracers lines
-        # lines: the tracer list of lines to search. options are section_tracer.warnings, section_tracer.errors, section_tracer.asserts, section_tracer.prints
-        # return: true if the line is found, otherwise false
-        def find_expected_line(expected_line, lines):
-            found_lines = [printInfo.message.strip() for printInfo in lines]
-            return expected_line in found_lines
-
-        def wait_for_critical_expected_line(expected_line, lines, time_out):
-            TestHelper.wait_for_condition(lambda : find_expected_line(expected_line, lines), time_out)
-            Report.critical_result(("Found expected line: " + expected_line, "Failed to find expected line: " + expected_line), find_expected_line(expected_line, lines))
+    @staticmethod
+    def wait_for_critical_expected_line(window, expected_message, print_infos, time_out):
+        TestHelper.wait_for_condition(lambda : TestHelper.find_expected_line(window, expected_message, print_infos), time_out)
+        Report.critical_result(("Found expected line: " + expected_message, "Failed to find expected line: " + expected_message), TestHelper.find_expected_line(window, expected_message, print_infos))
 
-        def wait_for_critical_unexpected_line(unexpected_line, lines, time_out):
-            TestHelper.wait_for_condition(lambda : find_expected_line(unexpected_line, lines), time_out)
-            Report.critical_result(("Unexpected line not found: " + unexpected_line, "Unexpected line found: " + unexpected_line), not find_expected_line(unexpected_line, lines))
+    @staticmethod
+    def wait_for_critical_unexpected_line(window, unexpected_line, print_infos, time_out):
+        TestHelper.wait_for_condition(lambda : TestHelper.find_expected_line(window, unexpected_line, print_infos), time_out)
+        Report.critical_result(("Unexpected line not found: " + unexpected_line, "Unexpected line found: " + unexpected_line), not TestHelper.find_expected_line(window, unexpected_line, print_infos))
```

Comment:
This is strange print "Unexpected line not found".

Intent appears to be that method waits until it either times out or a condition that represents error is logged.

wait_for_error_condition ? wait_for_terminating_output? Its hard to read wait_for_critical_unexpected_line as a method name and understand the intent IMHO